Adyen/Core/APIClient/BackoffScheduler.swift

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright (c) 2020 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+/// Decides what interval should be used to delay the next event, based on how many times the event has been fired before.
+public protocol IntervalCalculator {
+    func interval(for counter: UInt) -> DispatchTimeInterval
+}
+
+/// :nodoc:
+/// 2 seconds between the first 20 events, and then 10 seconds between the next 80 events,
+/// then `DispatchTimeInterval.never` is returned after that.
+public struct BackoffIntervalCalculator: IntervalCalculator {
```

Comment:
Where this rules comes from?
Does anyone actually going to wait payment confirmation for 15 minutes? :D 
maybe better something Fibonacci based?